.github/workflows/lint.yml

```diff
@@ -0,0 +1,44 @@
+name: lint
+
+on:
+  workflow_call:
+    inputs: {}
+  pull_request:
+    branches:
+    - main
+    - v*
+    - feature*
+    paths-ignore:
+    - CHANGELOG.md
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Install go
+      uses: actions/setup-go@v4
+      with:
+        # The versions of golangci-lint and setup-go here cross-depend and need to update together.
+        go-version: 1.21
+        # Either this action or golangci-lint needs to disable the cache
+        cache: false
+    - name: disarm go:embed directives to enable lint
+      continue-on-error: true # this fails if there are no go:embed directives
+      run: |
+        git grep -l 'go:embed' -- provider | xargs sed -i 's/go:embed/ goembed/g'
+    - name: prepare upstream
+      continue-on-error: true
```

Comment:
why continue on error here?

Reply:
Because the Makefile might not define an `upstream` target. If it does, then it is assumed to be necessary to compile the provider. If `make upstream` exists and fails, it is assumed the next step will fail.